### PR TITLE
feat: improve contract data autogen process

### PR DESCRIPTION
**Pull request by smartcontracts**
**Description**
Some minor tweaks to speed up the build/test process and also mainly adds a new `generate-deployed-artifacts.js` file which will automatically generate the contract data file that was previously written by hand by Matt.

Not a strictly necessary PR but it's useful and I had some extra time while on the plane.

### CODE REVIEW

File: packages/contracts/package.json [hunk 1]
```diff
@@ -7,51 +7,46 @@
     "dist/types/*.ts",
     "artifacts/contracts/**/*.json",
     "deployments/**/*.json",
-    "OVM",
-    "iOVM",
-    "libraries",
-    "mockOVM"
+    "chugsplash",
```
Comment (smartcontracts) [inline packages/contracts/package.json:10]: We forgot to update these as part of the previous cleanup PR that was just merged.

File: packages/contracts/package.json [hunk 2]
```diff
@@ -7,51 +7,46 @@
...
+    "L2",
+    "libraries"
   ],
   "types": "dist/index",
   "author": "Optimism PBC",
   "license": "MIT",
   "scripts": {
-    "all": "yarn clean && yarn build && yarn test && yarn lint:fix && yarn lint",
-    "build": "./scripts/build.sh",
-    "build:kovan": "yarn run build:contracts && yarn run build:typescript && yarn run build:copy && CHAIN_ID=69 yarn run build:dump && yarn run build:typechain",
```
Comment (smartcontracts) [inline packages/contracts/package.json:21]: I don't believe these were being used anymore.

File: packages/contracts/package.json [hunk 3]
```diff
@@ -7,51 +7,46 @@
...
+    "chugsplash",
+    "L1",
+    "L2",
+    "libraries"
   ],
   "types": "dist/index",
   "author": "Optimism PBC",
   "license": "MIT",
   "scripts": {
-    "all": "yarn clean && yarn build && yarn test && yarn lint:fix && yarn lint",
```
Comment (smartcontracts) [inline packages/contracts/package.json:19]: Not used anywhere.

File: packages/contracts/package.json [hunk 4]
```diff
@@ -7,51 +7,46 @@
...
     "build:contracts": "hardhat compile --show-stack-traces",
-    "build:dump": "ts-node \"bin/take-dump.ts\"",
-    "build:typechain": "hardhat typechain",
-    "test": "yarn run test:contracts",
+    "build:dump": "ts-node bin/take-dump.ts",
+    "autogen:markdown": "node scripts/generate-markdown.js",
+    "autogen:artifacts": "node scripts/generate-artifacts.js && node scripts/generate-deployed-artifacts.js",
+    "test": "yarn test:contracts",
     "test:contracts": "hardhat test --show-stack-traces",
-    "test:gas": "hardhat test \"test/contracts/OVM/execution/OVM_StateManager.gas-spec.ts\" --no-compile --show-stack-traces",
```
Comment (smartcontracts) [inline packages/contracts/package.json:29]: I don't think there's a need for this command.

File: packages/contracts/package.json [hunk 5]
```diff
@@ -7,51 +7,46 @@
...
-    "generate:artifacts": "node \"./scripts/generate-artifacts.js\"",
     "pre-commit": "lint-staged"
   },
   "dependencies": {
     "@eth-optimism/core-utils": "^0.5.2",
     "@ethersproject/abstract-provider": "^5.4.1",
     "@ethersproject/abstract-signer": "^5.4.1",
     "@ethersproject/contracts": "^5.4.1",
-    "@ethersproject/hardware-wallets": "^5.4.0",
-    "glob": "^7.1.6"
```
Comment (smartcontracts) [inline packages/contracts/package.json:54]: Glob is no longer a required dependency, we can make it a dev dep instead.

File: package.json [hunk 6]
```diff
@@ -36,7 +36,7 @@
     "typescript": "^4.3.5"
   },
   "scripts": {
-    "clean": "yarn lerna run clean",
+    "clean": "yarn lerna run clean --parallel",
```
Comment (smartcontracts) [inline package.json:39]: Doing clean in parallel saves about 7 seconds from the build time. Might as well do it.

File: packages/contracts/scripts/generate-deployed-artifacts.js [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+    })
+    .map((entry) => {
+      return entry.name
+    })
+
+  const artifactNames = []
+  for (const deploymentName of deploymentNames) {
+    const deploymentArtifacts = glob.sync(
+      path.resolve(__dirname, '../deployments') +
+        '/' +
```
Comment (tynes) [inline packages/contracts/scripts/generate-deployed-artifacts.js:32]: Perhaps `path.join` as this won't work on windows

Comment (smartcontracts) [reply packages/contracts/scripts/generate-deployed-artifacts.js:32]: Good point

**Comment by tynes** [comment]
This needs a rebase now

File: packages/contracts/package.json [hunk 8]
```diff
@@ -7,51 +7,46 @@
     "dist/types/*.ts",
     "artifacts/contracts/**/*.json",
     "deployments/**/*.json",
-    "OVM",
-    "iOVM",
-    "libraries",
-    "mockOVM"
+    "chugsplash",
+    "L1",
```
Comment (tynes) [inline packages/contracts/package.json:11]: Does this need to be `L1/**` or will it recursively work?

Comment (smartcontracts) [reply packages/contracts/package.json:11]: It works recursively

File: packages/contracts/scripts/generate-deployed-artifacts.js [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+
+  content += `
+  export const getDeployedContractArtifact = (name: string, network: string): any => {
+    return {
+      ${artifactNames
+        .map((artifactName) => {
+          return `${artifactName}: ${artifactName}`
+        })
+        .join(',\n')}
+    }[(network + '__' + name).replace(/-/g, '_')]
```
Comment (tynes) [inline packages/contracts/scripts/generate-deployed-artifacts.js:59]: This line is correct? Unclear if its supposed to be rendered

**Review by tynes — APPROVED** [review]

**Pull request merged**